packages/abstraxion-core/src/AbstraxionAuth.ts

```diff
@@ -339,18 +343,89 @@ export class AbstraxionAuth {
     this.triggerAuthStateChange(false);
   }
 
+  /**
+   * Checks if a grant is valid by verifying its expiration.
+   *
+   * @param {string} grantee - The address of the grantee.
+   * @param {string} granter - The address of the granter.
+   * @returns {Promise<boolean>} - Returns true if the grant is valid, otherwise false.
+   */
+  async checkGrantValidity(grantee: string, granter: string): Promise<boolean> {
+    if (!this.rpcUrl) {
+      throw new Error("AbstraxionAuth needs to be configured.");
+    }
+    if (!grantee) {
+      throw new Error("No keypair address");
+    }
+    if (!granter) {
+      throw new Error("No granter address");
+    }
+
+    const pollBaseUrl =
+      this.restUrl || (await fetchConfig(this.rpcUrl)).restUrl;
+
+    try {
+      const url = new URL(`${pollBaseUrl}/cosmos/authz/v1beta1/grants`);
+      url.search = new URLSearchParams({ grantee, granter }).toString();
+
+      const res = await fetch(url, { cache: "no-store" });
+      const data: GrantsResponse = await res.json();
+
+      if (data.grants.length === 0) {
+        console.warn("No grants found.");
```

Comment:
This is interesting, what situation did you have in mind where this is being used?

Reply:
I don't think we'll see it often, but wanted to cover the edge case. I'm not sure of this but maybe the API call could return an empty array if it's internal logic with the chain fails

Reply:
It will also happen when the authz grant expires